server/backend/api/app/crud/crud_user_role.py

```diff
@@ -72,78 +74,118 @@ def create_user_role(
+                    error_code=famConstants.ERROR_CODE_INVALID_REQUEST_PARAMETER,
+                    error_msg=error_msg,
+                )
+
+            if not forest_client_active(forest_client_validator_return):
+                error_msg = (
+                    f"Invalid role assignment request. Forest client number {forest_client_number} is not in active status:"
+                    + f"{get_forest_client_status(forest_client_validator_return)}"
+                )
+                raise_http_exception(
+                    error_code=famConstants.ERROR_CODE_INVALID_REQUEST_PARAMETER,
+                    error_msg=error_msg,
+                )
+
+            # Check if child role exists or add a new child role
+            child_role = find_or_create_forest_client_child_role(
+                db, forest_client_number, fam_role, requester
             )
+            # Role for associating with user
+            associate_role = child_role if require_child_role else fam_role
 
-        # Note: current FSA design in the 'request body' contains a
-        #     'forest_client_number' if it requires a child role.
-        child_role = find_or_create_forest_client_child_role(
-            db,
-            request.forest_client_number,
-            fam_role,
-            requester
+            # Create user/role assignment
+            handle_create_return = create_user_role_assignment(
+                db, fam_user, associate_role, requester
+            )
+            create_return_list.append(handle_create_return)
+    else:
+        # Create user/role assignment
+        handle_create_return = create_user_role_assignment(
+            db, fam_user, fam_role, requester
         )
+        create_return_list.append(handle_create_return)
 
-    # Role Id for associating with user
-    associate_role_id = child_role.role_id if require_child_role else fam_role.role_id
+    LOGGER.debug(f"User/Role assignment executed successfully: {create_return_list}")
+    return create_return_list
 
-    # Create user/role assignment.
-    fam_user_role_xref = get_use_role_by_user_id_and_role_id(db, fam_user.user_id, associate_role_id)
+
+def create_user_role_assignment(
+    db: Session, user: models.FamUser, role: models.FamRole, requester: str
+):
+    create_user_role_assginment_return = None
+    fam_user_role_xref = get_use_role_by_user_id_and_role_id(
+        db, user.user_id, role.role_id
+    )
+    xref_dict = {"application_id": role.application_id, "user": user, "role": role}
 
     if fam_user_role_xref:
-        LOGGER.debug(
-            "FamUserRoleXref already exists with id: " +
-            f"{fam_user_role_xref.user_role_xref_id}."
+        xref_dict = {**fam_user_role_xref.__dict__, **xref_dict}
+        error_msg = (
+            f"Role {fam_user_role_xref.role.role_name} already assigned to user {fam_user_role_xref.user.user_name}. "
+            + f"FamUserRoleXref already exists with id: {fam_user_role_xref.user_role_xref_id}"
```

Comment:
I feel maybe we can remove the second line from error_msg, probably useless to return to frontend since first line already tells all.

Reply:
I think frontend is using different error messages, not directly using the messages from backend. But I agree, I should start to write error messages that is good enough that can be used by frontend directly for the future.